src/main/scala/viper/gobra/translator/implementations/translator/BuiltInMembersImpl.scala

```diff
@@ -505,20 +493,23 @@ class BuiltInMembersImpl extends BuiltInMembers {
 
       case (CopyFunctionTag, Vector(t1, t2, _)) =>
         /**
-          * requires 0 < p && p < 1
-          * requires forall i int :: { dst[i] } (0 <= i && i < len(dst)) ==> acc(&dst[i], 1-p)
+          * requires 0 < p
+          * requires forall i int :: { dst[i] } (0 <= i && i < len(dst)) ==> acc(&dst[i], write)
           * requires forall i int :: { src[i] } (0 <= i && i < len(src)) ==> acc(&src[i], p)
-          * requires forall i int :: { dst[i] } (0 <= i && i < len(dst) && (forall j int :: 0 <= j && j < len(src) ==> &dst[i] != &src[j])) ==> acc(&dst[i], p)
           * ensures len(dst) <= len(src) ==> res == len(dst)
           * ensures len(src) < len(dst) ==> res == len(src)
-          * ensures forall i int :: { dst[i] } 0 <= i && i < len(dst) ==> acc(&dst[i], 1-p)
+          * ensures forall i int :: { dst[i] } 0 <= i && i < len(dst) ==> acc(&dst[i], write)
           * ensures forall i int :: { src[i] } 0 <= i && i < len(src) ==> acc(&src[i], p)
-          * ensures forall i int :: { dst[i] } (0 <= i && i < len(dst) && (forall j int :: 0 <= j && j < len(src) ==> &dst[i] != &src[j])) ==> acc(&dst[i], p)
-          * ensures forall i int :: { dst[i] } (0 <= i && i < len(src) && i < len(dst)) ==> dst[i] == old(src[i])
+          * ensures forall i int :: { dst[i] } (0 <= i && i < len(src) && i < len(dst)) ==> dst[i] == src[i]
```

Comment:
Is there a particular reason you dropped the ‚old‘? You loose all knowledge about the values of ‚src‘ if you pass all permissions to it to ‚copy‘